zebra-state/src/service/non_finalized_state/arbitrary.rs

```diff
@@ -51,7 +51,7 @@ impl Strategy for PreparedChain {
         if chain.is_none() {
             // Only generate blocks from the most recent network upgrade
             let mut ledger_state = LedgerState::default();
-            ledger_state.network_upgrade_override = None;
+            ledger_state.network_upgrade_override = Some(NetworkUpgrade::Nu5);
```

Comment:
This might cause some tests to fail until the PR is complete.

Reply:
If we do this 2 of our tests will fail: `finalized_equals_pushed` and `forked_equals_pushed`.

Error is: `internal error: entered unreachable code: older transaction versions only exist in finalized blocks, because of the mandatory canopy checkpoint`

https://github.com/ZcashFoundation/zebra/blob/27e197a12bb2391afd26ad243a84d3526b0af543/zebra-state/src/service/non_finalized_state/chain.rs#L196

Reply:
Yeah, you're right, that doesn't quite work like I expected. I'll push a fix.

Reply:
See b8cc9189